cmd/argocd/commands/headless/headless.go

```diff
@@ -210,18 +210,19 @@ func StartLocalServer(ctx context.Context, clientOpts *apiclient.ClientOptions,
 	}
 	appstateCache := appstatecache.NewCache(cache.NewCache(&forwardCacheClient{namespace: namespace, context: ctxStr, compression: compression, redisHaProxyName: clientOpts.RedisHaProxyName, redisName: clientOpts.RedisName}), time.Hour)
 	srv := server.NewServer(ctx, server.ArgoCDServerOpts{
-		EnableGZip:           false,
-		Namespace:            namespace,
-		ListenPort:           *port,
-		AppClientset:         appClientset,
-		DisableAuth:          true,
-		RedisClient:          redis.NewClient(&redis.Options{Addr: mr.Addr()}),
-		Cache:                servercache.NewCache(appstateCache, 0, 0, 0),
-		KubeClientset:        kubeClientset,
-		Insecure:             true,
-		ListenHost:           *address,
-		RepoClientset:        &forwardRepoClientset{namespace: namespace, context: ctxStr, repoServerName: clientOpts.RepoServerName},
-		EnableProxyExtension: false,
+		EnableGZip:                    false,
+		Namespace:                     namespace,
+		ListenPort:                    *port,
+		AppClientset:                  appClientset,
+		DisableAuth:                   true,
+		RedisClient:                   redis.NewClient(&redis.Options{Addr: mr.Addr()}),
+		Cache:                         servercache.NewCache(appstateCache, 0, 0, 0),
+		KubeClientset:                 kubeClientset,
+		Insecure:                      true,
+		ListenHost:                    *address,
+		RepoClientset:                 &forwardRepoClientset{namespace: namespace, context: ctxStr, repoServerName: clientOpts.RepoServerName},
+		EnableProxyExtension:          false,
+		DisableDefaultProjectCreation: true,
```

Comment:
I decided to disable the default project creation in headless use-case, as I felt it would be side-effect there. What do you think?

Reply:
I don't think that should happen. If running Argo CD in [headless mode](https://argo-cd.readthedocs.io/en/stable/operator-manual/core/#introduction), the expectation would be that there should not be any breaking changes (which this would be). 

IMO I think the way to go is to add a command line option here (defaulting to `false`).

On the other side, it totally makes sense for any command _invoking_ headless to have this set to `true` (since as you say, running e.g `argocd repo add` should not on its own create a `default` `AppProject` if it has been disabled in `argocd-server`.

Do you have any thoughts on this @crenshaw-dev?

Reply:
Hmm. The more I'm looking at this, I'm starting to wonder if it wouldn't be better if the `initDefaultProject` function was instead moved to the application controller, to better accommodate the headless use case.

Reply:
> IMO I think the way to go is to add a command line option here (defaulting to false).

Thanks for the suggestion! I've implemented your idea in [this commit](https://github.com/argoproj/argo-cd/pull/15530/commits/f5af4ad50d264a5c594406ce8114e314e0d6a29c)

Regarding moving initDefaultProject to the application controller, I'm open to it if it better suits headless use cases. 😊